src/status_im/transport/handlers.cljs

```diff
@@ -2,152 +2,66 @@
   ::init-whisper
-  (fn [{:keys [web3 public-key groups updates-public-key updates-private-key status contacts pending-messages]}]
-    (protocol/init-whisper!
+  (fn [{:keys [web3 public-key groups status contacts pending-messages]}]
+    (transport/init-whisper!
      {:web3                        web3
       :identity                    public-key
-      :groups                      groups
-      :callback                    #(re-frame/dispatch [:incoming-message %1 %2])
-      :ack-not-received-s-interval 125
-      :default-ttl                 120
-      :send-online-s-interval      180
-      :ttl-config                  {:public-group-message 2400}
-      :max-attempts-number         3
-      :delivery-loop-ms-interval   500
-      :profile-keypair             {:public  updates-public-key
-                                    :private updates-private-key}
-      :hashtags                    (mapv name (handlers/get-hashtags status))
-      :pending-messages            pending-messages
-      :contacts                    (keep (fn [{:keys [whisper-identity
-                                                      public-key
-                                                      private-key]}]
-                                           (when (and public-key private-key)
-                                             {:identity whisper-identity
-                                              :keypair  {:public  public-key
-                                                         :private private-key}}))
-                                         contacts)
-      :post-error-callback         #(re-frame/dispatch [::post-error %])})))
+      :pending-messages            pending-messages})))
 
-(defn get-message-id [status-message]
-  (web3.utils/sha3 status-message))
+;; TODO (yenda) add handlers for cutsom values https://github.com/cognitect/transit-cljs/wiki/Getting-Started
+(def reader (transit/reader :json))
+(def writer (transit/writer :json))
 
-(defn serialize-status-message [status-message]
-  (prn-str status-message))
+(defn serialize [o] (transit/write writer o))
+(defn deserialize [o] (try (transit/read reader o) (catch :default e nil)))
 
-(defn deserialize-status-message [status-message]
-  (reader/read-string status-message))
+(defn get-message-id [status-message]
+  (web3.utils/sha3 status-message))
 
 (defn parse-payload [message]
-  (let [{:keys [payload sig recipientPublicKey]}            (js->clj js-message :keywordize-keys true)
-        {:keys [protocol transport-message status-message]} (-> payload
-                                                                web3.utils/to-utf8
-                                                                reader/read-string)
-        {:keys [message-type acks]}                         transport-message]
+  (let [{:keys [payload sig]} (js->clj js-message :keywordize-keys true)
+        status-message        (-> payload
+                                  web3.utils/to-utf8)]
```

Comment:
@yenda Any particular reason why we are using js->clj to deserialize the raw message and then transit to deserialize the payload ? I'm little bit confused by that.